### PR TITLE
[Combobox] Add index to uniquely identify duplicate options.

### PR DESCRIPTION
**Pull request by chrstntdd**
Devs shouldn't be providing duplicate options for users to choose from, but with this we can at least keep trucking along without running into the issue described in #216

With this change, we could also remove the need for running the hashing function to track the active descendant, but I'll leave that up to you.

### CODE REVIEW

File: packages/combobox/src/index.js [hunk 1]
```diff
@@ -398,7 +400,9 @@ export const ComboboxInput = forwardRef(function ComboboxInput(
   const inputValue =
     autocomplete && (state === NAVIGATING || state === INTERACTING)
       ? // When idle, we don't have a navigationValue on ArrowUp/Down
-        navigationValue || controlledValue || value
+        (navigationValue && stripIndex(navigationValue)) ||
```
Comment (chrstntdd) [inline packages/combobox/src/index.js]: Heck, this function can just be inlined.

**Comment by chrstntdd** [comment]
This is also currently missing a call to `stripIndex` at the `SELECT_WITH_KEYBOARD` action.

**Comment by chrstntdd** [comment]
If this ends up being merged in - it would seem that the hashing function could be removed. Let me know!

**Comment by chancestrickland** [comment]
@ryanflorence If you have time to glance at this quickly, can you let me know if you see any flags? I tested this locally and it seems to be fine, but A) I can't imagine why we'd need to support multiple items with the same value in the first place, and B) I haven't 100% wrapped my head around everything going on in this component 😅

**Pull request closed without merging**